### PR TITLE
[release-v0.17] wallet2: wait for propagation timeout before marking tx as failed

### PR DESCRIPTION
**Pull request by xiphon**

### CODE REVIEW

File: src/wallet/wallet2.cpp [hunk 1]
```diff
@@ -2988,9 +2990,11 @@ void wallet2::update_pool_state(std::vector<std::tuple<cryptonote::transaction,
         LOG_PRINT_L1("Pending txid " << txid << " not in pool, marking as not in pool");
         pit->second.m_state = wallet2::unconfirmed_transfer_details::pending_not_in_pool;
       }
-      else if (pit->second.m_state == wallet2::unconfirmed_transfer_details::pending_not_in_pool && refreshed)
+      else if (pit->second.m_state == wallet2::unconfirmed_transfer_details::pending_not_in_pool && refreshed &&
+        now > std::chrono::system_clock::from_time_t(pit->second.m_timestamp) + tx_propagation_timeout)
```
Comment (moneromooo-monero) [inline src/wallet/wallet2.cpp]: m_sent_time is probably better here, though I dont' remember all the details... Both are time(NULL) at init time though.

**Review by moneromooo-monero — APPROVED** [review]

File: src/wallet/wallet2.cpp (hunk 1 above)
Comment (xiphon) [reply src/wallet/wallet2.cpp]: updated

**Review by selsta — APPROVED** [review]
Tested

**Review by fluffypony — APPROVED** [review]
Reviewed

**Pull request merged**